internal/binutils/binutils.go

```diff
@@ -140,7 +144,65 @@ func initTools(b *binrep, config string) {
 		b.addr2line, b.addr2lineFound = findExe("gaddr2line", append(paths["addr2line"], defaultPath...))
 	}
 	b.nm, b.nmFound = findExe("nm", append(paths["nm"], defaultPath...))
-	b.objdump, b.objdumpFound = findExe("objdump", append(paths["objdump"], defaultPath...))
+	// Look for llvm-objdump and objdump.
+	b.objdump, b.isLLVMObjdump, b.objdumpFound = findObjdump(append(paths["objdump"], defaultPath...))
+}
+
+// findObjdump finds and returns preferred objdump binary.
+// Order of preference is: llvm-objdump, objdump.
+// On MacOS only, also looks for gobjdump with least preference.
+func findObjdump(paths []string) (string, bool, bool) {
+	objdumpNames := []string{"llvm-objdump", "objdump"}
+	if runtime.GOOS == "darwin" {
+		objdumpNames = append(objdumpNames, "gobjdump")
+	}
+
+	for _, objdumpName := range objdumpNames {
+		if objdump, objdumpFound := findExe(objdumpName, paths); objdumpFound {
+			cmdOut, err := exec.Command(objdump, "--version").Output()
+			if err != nil {
+				continue
+			}
+			if llvmObjdumpFound := isLLVMObjdump(string(cmdOut)); llvmObjdumpFound {
+				return objdump, llvmObjdumpFound, llvmObjdumpFound
+			}
+			if buObjdumpFound := isBuObjdump(string(cmdOut)); buObjdumpFound {
+				return objdump, false, buObjdumpFound
+			}
+		}
+	}
+	return "", false, false
+}
+
+func isLLVMObjdump(output string) bool {
+	fields := objdumpLLVMVerRE.FindStringSubmatch(output)
+	if len(fields) != 6 {
+		// Check if built from trunk.
+		output = strings.ToLower(output)
+		return strings.Contains(output, "trunk") && strings.Contains(output, "llvm version")
```

Comment:
I would change the regular expression to something like `(?i)LLVM) version (?:(\d*)\.(\d*)\.(\d*)|.*(trunk).*)`. This would simplify this code:
* No need for ToLower and string.Contains(llvm output) check.
* If capture group 4 is "trunk" then it's trunk.
* Otherwise capture group 1, 2 and 3 must be numbers and is the version number.

Reply:
You can drop `ToLower` or `(?i)` for `llvm`. I have fixed the internal issue.

Reply:
done.